cluster/examples/kubernetes/ceph/create-external-cluster-resources.sh

```diff
@@ -56,6 +56,46 @@ function createCephCSIKeyringCephFSProvisioner() {
   echo "export CSI_CEPHFS_PROVISIONER_SECRET=$cephCSIKeyringCephFSProvisionerKey"
 }
 
+function getCephExternalFSID() {
+  cephFSID=$(ceph fsid)
+  echo "export ROOK_EXTERNAL_FSID=$cephFSID"
+}
+
+function getCephExternalMonData() {
+  cephQuorumOp="$(ceph quorum_status)"
+  # fetch the quorum leader name
+  cephQuorumLeaderName="$(echo $cephQuorumOp |sed -n 's@.*"quorum_leader_name":"\([^"]*\)".*@\1@gp')"
+  # get the <ip_address>:<port> of the quorum leader
+  # (and remove any '/<digit>' from the tail end of the output)
+  cephQuorumLeaderIP="$(echo $cephQuorumOp |sed -n -r 's@.*"name":"'$cephQuorumLeaderName'",(.*)@\1@gp' |sed -r -n 's@"public_addr":"([^"]*)".*|.@\1@gp' |sed 's@/.*@@g')"
```

Comment:
Personally, I wouldn't try to read json output in bash, it seems a bit "weak", I'd go with a json parser like `jd` but it can be challenging since it will come as a requirement to run that script.
Now I think I'm leaning forward re-writing this in python so it's more maintainable/testable and would ease the json marshaling/unmarshalling.

Reply:
@leseb ,

This brings an extra dependency on `python` package itself. Currently in our `rook` packaging / source we don't have any python files and has ZERO dependency (on python).

IMHO, Python is  divided, by that I mean, whether to write things in python 2.x or 3.x and we have to check for the backward compatibility of the script. By design python 3.x is not backward compatible with 2.x.

Current shell script works fine, if the output format is fixed. Taking a look into it, if we can confirm that (fixed output format) the current way should be fine.

Another way is to, write our own utility binary in Golang and ship a static binary (which works on all flavours of amd64 linux and there should not be any external dependencies). Use the utility binary in the (shell) script to get the json parsed values.

Reply:
Python 3 only would be fine. The issue with the binary blob is that it's not introspectable, the script is also meant to act as an example/material/guidance on what to do. If you ship a binary Admins are blind.
Currently, as is the script is not as simple as I would like it to be :-/